cstore_reader.c

```diff
@@ -1144,10 +1183,108 @@ DeserializeBlockData(StripeBuffers *stripeBuffers, uint64 blockIndex,
 			/* store current block's data buffer to be freed at next block read */
 			blockData->valueBuffer = valueBuffer;
 		}
+		else if (columnAdded)
+		{
+			/* populate exists and value buffers for missing column */
+			int rowIndex = 0;
+			bool useDefaultValue = attributeForm->atthasdef;
+			Datum defaultValue = (Datum) 0;
+			char *columnName = attributeForm->attname.data;
+
+			if (useDefaultValue)
+			{
+				useDefaultValue = DefaultValueForColumn(tupleDescriptor->constr,
+														columnIndex, columnName,
+														&defaultValue);
+			}
+
+			for (rowIndex = 0; rowIndex < rowCount; rowIndex++)
+			{
+				blockData->existsArray[rowIndex] = useDefaultValue;
+				/* it is safe to assign datum value since this array content
+				 * is never deallocated.
+				 */
```

Comment:
Comment doesn't match our coding styles. The text of comment should start from 2nd line and the it should start with a capital letter.

```C
/*
 * It is safe to assign datum value since this array content
 * is never deallocated.
 */
```